blueprints/serverless/cloud-run-explore/README.md

```diff
@@ -0,0 +1,158 @@
+
+Be aware that in this case the Cloud Run service can also be reached through the default URL. To limit access only through the custom domain see the next use case.
+
+### Use case 3: Cloud Run service exposed only via custom domain
+To block access to the default URL, you can configure Ingress Settings so that Internet requests will be accepted only if they come through the Load Balancer:
+
+<p align="center"> <img src="use-case-3.png" width="700"> </p>
+
+You only need to set one more value in the previous `terraform.tfvars` file:
+```
+project_id       = [your-project-id]
+custom_domain    = "cloud-run-explore.example.org"
+ingress_settings = "internal-and-cloud-load-balancing"
+```
+
+The default URL is still created but if you try to visit it, you should see a forbidden error:
+<p align="center"> <img src="forbidden.png" width="700"> </p>
+
+### Use case 4: Cloud Run service protected by Cloud Armor
+To use Cloud Armor to protect the Cloud Run service, you need to create a security policy to enforce in the load balancer:
+<p align="center"> <img src="use-case-4.png" width="700"> </p>
+
+The code allows to block a list of IPs and a specific URL path. For example, you may want to block access to a login page to external users. To test its behavior, by default all IPs and the path `"/login.html"` are blocked, but you can override any of these settings with your own values:
+```
+project_id       = [your-project-id]
+custom_domain    = "cloud-run-explore.example.org"
+ingress_settings = "internal-and-cloud-load-balancing"
+security_policy = {
+  enabled      = true
+  ip_blacklist = ["79.149.0.0/16"]
+  path_blocked = "/admin.html"
+}
+```
+
+Note that to avoid users to bypass the Cloud Armor policy you need to block access through the default URL. Ingress settings is configured to do that.
+
+### Use case 5: Cloud Run service protected by Cloud Armor and Identity-Aware Proxy
+You can enable IAP at the load balancer to control access using identity and context:
+<p align="center"> <img src="use-case-5.png" width="700"> </p>
+Use your own email as identity to access the Cloud Run service:
+
+```
+project_id       = [your-project-id]
+custom_domain    = "cloud-run-explore.example.org"
+ingress_settings = "internal-and-cloud-load-balancing"
+security_policy = {
+  enabled      = true
+  ip_blacklist = ["79.149.0.0/16"]
+}
+iap = {
+  enabled = true
+  email   = "user@example.org"
+}
+```
+When visiting it you may be redirected to login with Google. You can use an incognito window to test this behavior.
+
+## Cleaning up your environment
+The easiest way to remove all the deployed resources is to run the following command:
+```
+terraform destroy
+```
+The above command will delete the associated resources so there will be no billable charges made afterwards. IAP Brands, though, can only be created once per project and not deleted. Destroying a Terraform-managed IAP Brand will remove it from state but will not delete it from Google Cloud.
```

Comment:
add variables and outputs block here.

I'd also suggest a test block similar [to this one](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/REFERENCES.md) to enable at a least a basic test of the blueprint.